tools/aie2xclbin/XCLBinGen.cpp

```diff
@@ -576,18 +586,77 @@ static std::string chesshack(const std::string &input) {
   return result;
 }
 
+// A pass which removes the alignment attribute from llvm load operations, if
+// the alignment is 2.
+//
+// Example replaces:
+//
+// ```
+//  %113 = llvm.load %112 {alignment = 2 : i64} : !llvm.ptr -> vector<32xbf16>
+// ```
+//
+// with
+//
+// ```
+//  %113 = llvm.load %112 : !llvm.ptr -> vector<32xbf16>
+// ```
+//
+// If this pass is not included in the pipeline, there is an alignment error
+// later in the compilation. This is a temporary workaround while a better
+// solution is found: propagation of memref.assume_alignment is one option. See
+// also https://jira.xilinx.com/projects/AIECC/issues/AIECC-589
+namespace {
+struct RemoveAlignment2FromLLVMLoadPass
+    : public PassWrapper<RemoveAlignment2FromLLVMLoadPass,
+                         OperationPass<ModuleOp>> {
+  void runOnOperation() override {
+    getOperation().walk([](Operation *op) {
+      if (auto loadOp = dyn_cast<LLVM::LoadOp>(op)) {
+        auto alignmentAttr = loadOp.getAlignmentAttr();
+        if (alignmentAttr && alignmentAttr.getValue() == 2)
+          loadOp.setAlignment(std::optional<uint64_t>());
+      }
```

Comment:
Eventually, we might have to do the same for `LLVM::StoreOp`. No change this now, with a little bit of luck it will be fixed for good and we won't need it any more, but we should make a note in case this lives longer than we wish and the "misaligned" stores start triggering the issue again.

Reply:
Good point, will keep in mind. I'll land this PR as is as soon as it passes CI.